vllm/engine/async_llm_engine.py

```diff
@@ -264,11 +317,11 @@ def start_background_loop(self) -> None:
         """Start the background loop."""
         if self.is_running:
             raise RuntimeError("Background loop is already running.")
-        self.background_loop = asyncio.get_event_loop().create_task(
-            self.run_engine_loop())
-        self.background_loop.add_done_callback(
-            partial(_raise_exception_on_finish,
-                    request_tracker=self.request_tracker))
+        self._request_tracker.init_event()
+        self.background_loop = self._task_set.run(
```

Comment:
if this `task_set` will only have one task, why is a "set" needed? Can we just shield the background loop?